### PR TITLE
New tip for jstips - Improve Nested Conditionals

### PR DESCRIPTION
**Pull request by AlbertoFuente**

### CODE REVIEW

**Comment by padsbanger** [comment]
Using `switch` in Javascript is not a good pattern. https://toddmotto.com/deprecating-the-switch-statement-for-object-literals/

**Comment by AlbertoFuente** [comment]
Thanks you for your point padsbanger, I know it's not appropriate to use switch on javascript and the problems associated with its use, I'm trying to convey the steps to improve nested if and nested if with several checks in each statement. Maybe I should add the problems associated with the use of switch.

**Comment by AlbertoFuente** [comment]
In line with this, I remember that Eric Elliot retracted his advice about not using switch statements https://twitter.com/_ericelliott/status/673011325568610305

**Comment by loverajoel** [comment]
@AlbertoFuente hey nice tip, it's true the switch is not a good pattern but you can explain it.
I like the tip, if you edit your tip I will merge this the January 3, please add your tip at the end of the [Tip List](https://github.com/loverajoel/jstips#tips-list) and update the date.
Thanks for your help.

File: README.md [hunk 1]
```diff
@@ -43,6 +43,88 @@ items.splice(2, 0, 'hello');
...
+var colorObj = {
+  'black': printBlackBackground,
+  'red': printRedBackground,
+  'blue': printBlueBackground,
+  'green': printGreenBackground,
+  'yellow': printYellowBackground
+};
+
+if (color) {
+  colorObj[color]();
```
Comment (manmadareddy) [inline README.md]: It will be a good idea to check
colorObj.hasOwnProperty(color) 
before we use.

**Comment by AlbertoFuente** [comment]
@loverajoel Ok, I will improve it, and thank you for this great initiative. :+1:
@manmadareddy Thank you for your contribution, I will add to the example. :+1:

**Comment by ghost** [comment]
``` javascript
switch(!!color) {
  case (typeof color != 'string'): break;
  case (color == 'blue'): console.log('Blue') break;
  case (color == 'green'): console.log('Green') break;
}
```

**Comment by kotAPI** [comment]
we want more jstips!!! one jstip a day keps bugs away!

**Comment by AlbertoFuente** [comment]
Thanks for improvement @noncentz.

**Comment by loverajoel** [comment]
@AlbertoFuente the contributions guidelines changed, and the new tips will be at the top of the list.
Please can you move your tip to the top and squash your commits? 
Thanks

**Comment by AlbertoFuente** [comment]
I've already put the tip above the others, I have a question, I see different types of numbering for tips:

# 0 - Insert ...

# 1 - angularjs ...

# 02 - ReactJs ...

It would be nice to specify a standard for numbering the tips.

**Comment by loverajoel** [comment]
@AlbertoFuente awesome! the tips numbering should be 00, 01, 02, 03...  I made a mistake with the previous two and now I can't fix because the tweets are going to lost the link(works with the title).
Please add the 03 and don't forget squash your commits.
After that I'll merge the PR.

Thanks man

**Comment by AlbertoFuente** [comment]
I'm having troubles with the squash

**Comment by loverajoel** [comment]
@AlbertoFuente this is a good article about [squash commits](https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/).
Don't worry I can do the squash for you after merger the PR

**Comment by AlbertoFuente** [comment]
Thank you very much, there's something I'm not doing well.

**Comment by loverajoel** [comment]
@AlbertoFuente don't worry, thanks for your tip!

**Review by europrime — APPROVED** [review]

**Pull request merged**